src/systemctl/systemctl-edit.c

```diff
@@ -542,31 +602,32 @@ int verb_edit(int argc, char *argv[], void *userdata) {
                         return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Cannot edit %s: unit is masked.", *tmp);
         }
 
-        r = find_paths_to_edit(bus, names, &paths);
+        r = find_paths_to_edit(bus, names, &edit_files);
         if (r < 0)
                 return r;
-
-        if (strv_isempty(paths))
+        if (!edit_files)
                 return -ENOENT;
 
-        r = run_editor(paths);
+        r = run_editor(edit_files);
         if (r < 0)
                 goto end;
 
-        STRV_FOREACH_PAIR(original, tmp, paths) {
+        for (EditFile *f = edit_files; f && f->path; f++) {
```

Comment:
Not a matter, but the condition `f` can be dropped, as it is already checked in the above.

Reply:
Also `const Edit *f`.

Reply:
Can't be const, we modify `tmp` in the loop a few lines down